src/metabase/api/automagic_dashboards.clj

```diff
@@ -115,8 +144,9 @@
    show   Show
    prefix Prefix
    rule   Rule}
-  (-> entity-id-or-query ((->entity entity)) (automagic-analysis {:show (keyword show)
-                                                                  :rule ["table" prefix rule]})))
+  (-> (->entity entity entity-id-or-query)
+      (automagic-analysis {:show (keyword show)
+                           :rule ["table" prefix rule]})))
```

Comment:
As a result of the multimethod change above I think the rest of the code here reads a lot cleaner.